### PR TITLE
[2] refactor: add data-cy and classname props

### PR DESCRIPTION
**Pull request by nataliepina**
# Description

Throughout these changes I am looking to achieve the following:
- Add `data-cy` attributes that we will be able to pass in custom values. Improving DX with our components + Cypress testing. I have opted for consistent use of `"data-cy"` instead of `["data-cy"]` as the [] is unnecessary. 
- Add `className` to an outermost container on components, that we can use for style overrides in the future.
- Consistency in the casing of prop definitions, as this displays to the Storybook UI in the Docs tab.
- Remove unnecessary `defaultProps` and opt for default values set during prop destructuring. 

## Checklist

- [ ] This PR is associated with a JIRA and it is mentioned in the commit message footer ("Closes …")
- [ ] Significant changes have been tested downstream to avoid breaking changes
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [ ] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

**Review by mstruebing — APPROVED** [review]
👍

File: packages/modal/components/ModalBase.tsx [hunk 1]
```diff
@@ -25,34 +25,58 @@ export enum ModalSizes {
...
   children?: React.ReactNode;
-  /** Controls whether the modal animates in and out. ⚠️Do not use this directly⚠️ */
+  /**
+   * Controls whether the modal animates in and out
+   * ⚠️ Do not use this directly
+   */
   isAnimated?: boolean;
-  /** Whether the modal is open */
+  /**
+   * Wether the modal is open or not
```
Comment (seialkali) [inline packages/modal/components/ModalBase.tsx:34]: 🥜 `Wether` should be `Whether`

File: packages/modal/components/ModalContents.tsx [hunk 2]
```diff
@@ -3,15 +3,32 @@ import { cx } from "@emotion/css";
 import { modalWrapper } from "../style";
 import { flex } from "../../shared/styles/styleUtils";
 export interface ModalContentsProps {
+  /**
+   * Wether the modal is open or not
```
Comment (seialkali) [inline packages/modal/components/ModalContents.tsx:7]: 🥜 Should be spelled `Whether` here too

**Review by seialkali — APPROVED** [review]
Noticed a couple of spelling errors, but they're NBD!

**Pull request merged**